src/main/java/org/truffleruby/language/control/CheckIfPatternsMatchedNode.java

```diff
@@ -17,14 +18,16 @@
 public final class CheckIfPatternsMatchedNode extends RubyContextSourceNode {
 
     @Child RubyNode inspected;
+    @Child ToJavaStringNode toJavaStringNode;
 
     public CheckIfPatternsMatchedNode(RubyNode inspected) {
         this.inspected = inspected;
     }
 
     @Override
     public Object execute(VirtualFrame frame) {
-        String message = inspected.execute(frame).toString();
+        toJavaStringNode = ToJavaStringNode.create(inspected);
```

Comment:
We must not create nodes in PE code. So it should `toJavaStringNode = ToJavaStringNode.create();` in the constructor.